File: game.py
import discord
from discord.ext import commands
from discord import app_commands
import json
import os
import aiohttp
import asyncio
from datetime import datetime
from typing import Optional, List, Dict
import logging

logger = logging.getLogger(__name__)

# ========== AI CONFIGURATION ==========
# You can plug in OpenAI, Local LLM, or free API here
AI_ENABLED = False  # Set to True when you have API key
AI_API_KEY = os.getenv('AI_API_KEY', '')
AI_API_URL = os.getenv('AI_API_URL', 'https://api.openai.com/v1/chat/completions')

# ========== GAME DATA FILE ==========
DATA_FILE = 'data/players.json'

def load_players():
    """Load player data from JSON file"""
    if not os.path.exists(DATA_FILE):
        os.makedirs('data', exist_ok=True)
        return {}
    with open(DATA_FILE, 'r') as f:
        return json.load(f)

def save_players(data):
    """Save player data to JSON file"""
    with open(DATA_FILE, 'w') as f:
        json.dump(data, f, indent=2)

# ========== EPISODE DATA ==========
EPISODES = {
    1: {
        'name': 'Setting Sail',
        'description': 'The journey for the greatest treasure begins at Dawn Island docks.',
        'scenes': [
            {
                'id': 'meet_boat_builder',
                'text': "**🛠️ Boat Builder:** *'Captain! The ship is ready. But where are we heading?'*",
                'choices': [
                    {'text': '🏴‍☠️ "We hunt the One Piece!"', 'next_scene': 'fish_kingdom_warning', 'stat_change': {'bounty': 10000}},
                    {'text': '🌊 "To the Grand Line!"', 'next_scene': 'fish_kingdom_warning', 'stat_change': {'bounty': 5000}},
                    {'text': '🗺️ "First, we need a navigator."', 'next_scene': 'wait_navigator', 'stat_change': {}}
                ]
            },
            {
                'id': 'fish_kingdom_warning',
                'text': "**🐟 Fish Soldier emerges from water:** *'Turn back, human! The Fish Kingdom claims these waters!'*\n\n⚔️ **COMBAT INITIATED!**",
                'choices': [
                    {'text': '💪 "Gomu Gomu no Punch!" (Use rubber powers)', 'next_scene': 'victory', 'stat_change': {'bounty': 25000}},
                    {'text': '🥷 "Ninja, handle this!"', 'next_scene': 'victory', 'stat_change': {'bounty': 15000}},
                    {'text': '🏃 "Retreat to the ship!"', 'next_scene': 'retreat', 'stat_change': {}}
                ]
            },
            {
                'id': 'victory',
                'text': "**VICTORY!** The Fish Soldiers retreat to the depths.\n\n*'The Fish Kingdom knows we're coming...'*",
                'choices': [
                    {'text': '➡️ Continue to Episode 2', 'next_scene': 'episode_end', 'stat_change': {}}
                ]
            },
            {
                'id': 'retreat',
                'text': "You escape to open sea. The Fish Kingdom now knows your face.\n\n*'We'll need a better plan...'*",
                'choices': [
                    {'text': '➡️ Regroup and continue', 'next_scene': 'episode_end', 'stat_change': {}}
                ]
            },
            {
                'id': 'wait_navigator',
                'text': "**Boat Builder:** *'Smart thinking, Captain. I know a town nearby...'*",
                'choices': [
                    {'text': '➡️ Sail to the town', 'next_scene': 'episode_end', 'stat_change': {}}
                ]
            },
            {
                'id': 'episode_end',
                'text': "**📺 END OF EPISODE 1**\n\n*To be continued...*\n\nThe crew sails toward the unknown, the Fish Kingdom in pursuit, and somewhere out there... the Map Drawer waits.",
                'choices': [],
                'is_end': True
            }
        ]
    },
    2: {
        'name': 'The Open Sea',
        'description': 'Days pass on the endless blue. Something approaches on the horizon...',
        'scenes': [
            {
                'id': 'open_ocean',
                'text': "**⭐ You:** *'Land? Anyone?'*\n\n**🛠️ Boat Builder:** *'Nothing yet, Captain. But the sea feels... strange.'*\n\nThe water churns. Something large swims beneath.",
                'choices': [
                    {'text': '🔍 "Everyone on alert!"', 'next_scene': 'sea_king', 'stat_change': {}},
                    {'text': '🎣 "Let\'s fish. I\'m hungry."', 'next_scene': 'fish_catches', 'stat_change': {}}
                ]
            },
            {
                'id': 'sea_king',
                'text': "**🐉 A SEA KING erupts from the water!**\n\n*'ROOOOAR!'* It's massive – bigger than your ship!\n\n**🥷 Ninja:** *'Captain! What do we do?!'*",
                'choices': [
                    {'text': '💪 Stretch and punch it!', 'next_scene': 'sea_king_escape', 'stat_change': {'bounty': 50000}},
                    {'text': '🏃 Outrun it!', 'next_scene': 'sea_king_escape', 'stat_change': {}},
                    {'text': '🤝 Try to communicate', 'next_scene': 'sea_king_friendly', 'stat_change': {}}
                ]
            },
            {
                'id': 'sea_king_escape',
                'text': "After a desperate struggle, you escape! The Sea King dives back into the depths.\n\n**⭐ You:** *'That was too close... We need to get stronger.'*\n\nA small island appears on the horizon. Smoke rises from a town.",
                'choices': [
                    {'text': '🏝️ Head to the island', 'next_scene': 'episode_end', 'stat_change': {}}
                ]
            },
            {
                'id': 'sea_king_friendly',
                'text': "**🐉 Sea King:** *'...You're not afraid of me?'*\n\nIt speaks! This Sea King is intelligent. It reveals the location of a hidden island – where a legendary Map Drawer lives.",
                'choices': [
                    {'text': '🗺️ "Take us there!"', 'next_scene': 'episode_end', 'stat_change': {'bounty': 30000}}
                ]
            },
            {
                'id': 'fish_catches',
                'text': "**⭐ You:** *'Not bad, Boat Builder!'*\n\n**🐟 Fish Soldier (in the catch):** *'...You caught me. Please don't eat me. I can tell you about the Fish Kingdom...'*",
                'choices': [
                    {'text': '🗣️ "Talk. Now."', 'next_scene': 'fish_intel', 'stat_change': {}},
                    {'text': '🍳 "Dinner is dinner."', 'next_scene': 'episode_end', 'stat_change': {'bounty': -5000}}
                ]
            },
            {
                'id': 'fish_intel',
                'text': "**🐟 Fish Soldier:** *'The Fish King wants the One Piece. He has half a map. Your crew has the other half – with the Map Drawer... but you haven't found her yet. Find her before he does!'*",
                'choices': [
                    {'text': '🏃 "We sail faster!"', 'next_scene': 'episode_end', 'stat_change': {}}
                ]
            },
            {
                'id': 'episode_end',
                'text': "**📺 END OF EPISODE 2**\n\n*To be continued...*\n\nThe Map Drawer is out there. The Fish Kingdom is closer than ever.",
                'choices': [],
                'is_end': True
            }
        ]
    }
}

# ========== SLASH COMMAND COG ==========
class GameCog(commands.Cog):
    """One Piece Game Commands"""
    
    def __init__(self, bot):
        self.bot = bot
        self.players = load_players()
    
    @app_commands.command(name="sail", description="⭐ Begin your One Piece adventure")
    async def sail(self, interaction: discord.Interaction):
        """Start the game"""
        user_id = str(interaction.user.id)
        
        if user_id in self.players:
            embed = discord.Embed(
                title="🏴‍☠️ ALREADY SAILING",
                description=f"{interaction.user.display_name}, your journey has already begun!",
                color=0xFFA500
            )
            embed.add_field(name="Current Episode", value=f"Episode {self.players[user_id]['episode']}", inline=True)
            embed.add_field(name="Bounty", value=f"{self.players[user_id]['bounty']:,} Berries", inline=True)
            await interaction.response.send_message(embed=embed)
            return
        
        # Create new player
        self.players[user_id] = {
            'name': interaction.user.display_name,
            'episode': 1,
            'scene': 'meet_boat_builder',
            'bounty': 0,
            'crew': {
                'boat_builder': True,
                'ninja': False,
                'map_drawer': False
            },
            'rubber_powers': True,
            'completed': [],
            'started_at': datetime.now().isoformat()
        }
        save_players(self.players)
        
        embed = discord.Embed(
            title="⭐ YOUR ADVENTURE BEGINS ⭐",
            description=f"**{interaction.user.display_name}** has eaten the **Gomu Gomu no Mi** (Rubber Fruit) and set sail!",
            color=0xFF4500
        )
        embed.add_field(name="Crew", value="⭐ You (Rubber Human)\n🛠️ Boat Builder\n❌ Ninja (missing)\n❌ Map Drawer (missing)", inline=False)
        embed.add_field(name="Location", value="Dawn Island Docks", inline=True)
        embed.add_field(name="Episode", value="1 - Setting Sail", inline=True)
        embed.set_footer(text="Use /continue to begin your story!")
        
        await interaction.response.send_message(embed=embed)
        logger.info(f"New player: {interaction.user.display_name}")
    
    @app_commands.command(name="crew", description="⭐ View your current crew")
    async def crew(self, interaction: discord.Interaction):
        """Show crew status"""
        user_id = str(interaction.user.id)
        
        if user_id not in self.players:
            await interaction.response.send_message("❌ You haven't started yet! Use `/sail` first.")
            return
        
        p = self.players[user_id]
        
        embed = discord.Embed(
            title=f"⭐ {p['name']}'S CREW",
            color=0x00BFFF
        )
        
        crew_status = []
        crew_status.append("✅ **Boat Builder** (Shipwright)")
        crew_status.append("✅ **You** (Rubber Fruit User)")
        crew_status.append("❌ **Ninja** (Not recruited yet)")
        crew_status.append("❌ **Map Drawer** (Episode 3)")
        
        embed.add_field(name="👥 Crew Members", value="\n".join(crew_status), inline=False)
        embed.add_field(name="💰 Bounty", value=f"{p['bounty']:,} Berries", inline=True)
        embed.add_field(name="📺 Episode", value=p['episode'], inline=True)
        embed.add_field(name="🏝️ Location", value="Grand Line - Unknown Seas", inline=True)
        
        await interaction.response.send_message(embed=embed)
    
    @app_commands.command(name="continue", description="📺 Continue the story")
    async def continue_story(self, interaction: discord.Interaction):
        """Play next scene"""
        user_id = str(interaction.user.id)
        
        if user_id not in self.players:
            await interaction.response.send_message("❌ Use `/sail` to start your adventure first!")
            return
        
        player = self.players[user_id]
        current_ep = player['episode']
        current_scene_id = player.get('scene', 'meet_boat_builder')
        
        # Get episode data
        ep_data = EPISODES.get(current_ep)
        if not ep_data:
            await interaction.response.send_message("🌊 **To be continued...** More episodes are being written!")
            return
        
        # Find current scene
        current_scene = None
        for scene in ep_data['scenes']:
            if scene['id'] == current_scene_id:
                current_scene = scene
                break
        
        if not current_scene:
            current_scene = ep_data['scenes'][0]
        
        # Build embed
        embed = discord.Embed(
            title=f"📺 EPISODE {current_ep} - {ep_data['name']}",
            description=current_scene['text'],
            color=0xFF4500
        )
        
        if current_scene['choices']:
            choice_text = []
            for i, choice in enumerate(current_scene['choices'], 1):
                choice_text.append(f"{i}️⃣ {choice['text']}")
            embed.add_field(name="⚡ YOUR CHOICE", value="\n".join(choice_text), inline=False)
            embed.set_footer(text="Reply with the number of your choice (1, 2, or 3)")
            
            # Store awaiting choice
            self.bot.awaiting_choice[user_id] = {
                'scene': current_scene,
                'episode': current_ep
            }
        else:
            embed.set_footer(text="End of episode. /continue for next episode...")
        
        await interaction.response.send_message(embed=embed)
    
    @app_commands.command(name="bounty", description="💰 Check your bounty")
    async def bounty(self, interaction: discord.Interaction):
        """Show current bounty"""
        user_id = str(interaction.user.id)
        
        if user_id not in self.players:
            await interaction.response.send_message("❌ Start with `/sail` first!")
            return
        
        bounty = self.players[user_id]['bounty']
        
        if bounty < 100000:
            rank = "🪙 Rookie Pirate"
            color = 0x808080
        elif bounty < 500000:
            rank = "⚔️ Super Rookie"
            color = 0x00BFFF
        elif bounty < 1000000:
            rank = "💀 Warlord Tier"
            color = 0x800080
        else:
            rank = "👑 Emperor Level"
            color = 0xFFD700
        
        embed = discord.Embed(
            title="💰 BOUNTY POSTER 💰",
            description=f"**WANTED: {self.players[user_id]['name']}**\n\n{bounty:,} BERRIES",
            color=color
        )
        embed.add_field(name="Rank", value=rank, inline=True)
        embed.add_field(name="Devil Fruit", value="Gomu Gomu no Mi (Rubber)", inline=True)
        embed.set_footer(text="The World Government is watching...")
        
        await interaction.response.send_message(embed=embed)
    
    @app_commands.command(name="help", description="🏴‍☠️ Show all commands")
    async def help_cmd(self, interaction: discord.Interaction):
        """Help menu"""
        embed = discord.Embed(
            title="🏴‍☠️ ONE PIECE BOT - COMMANDS 🏴‍☠️",
            description="Your journey to find the One Piece begins here!",
            color=0xFFD700
        )
        embed.add_field(name="/sail", value="Begin your adventure", inline=False)
        embed.add_field(name="/crew", value="View your crew", inline=False)
        embed.add_field(name="/continue", value="Continue the story", inline=False)
        embed.add_field(name="/bounty", value="Check your bounty", inline=False)
        embed.add_field(name="/help", value="Show this menu", inline=False)
        embed.set_footer(text="⚡ Episode 3: The Map Drawer - Coming Soon!")
        
        await interaction.response.send_message(embed=embed)
    
    @app_commands.command(name="episodes", description="📺 List all episodes")
    async def episodes(self, interaction: discord.Interaction):
        """Show episode list"""
        embed = discord.Embed(
            title="📺 ONE PIECE - EPISODE LIST",
            description="Your adventure so far...",
            color=0xFF6347
        )
        
        for ep_num, ep_data in EPISODES.items():
            status = "🟢 Available" if ep_num <= self.players.get(str(interaction.user.id), {}).get('episode', 1) else "🔒 Coming Soon"
            embed.add_field(
                name=f"Episode {ep_num}: {ep_data['name']}",
                value=f"{ep_data['description']}\n{status}",
                inline=False
            )
        
        embed.set_footer(text="Episode 3 introduces the Map Drawer!")
        await interaction.response.send_message(embed=embed)

# ========== SETUP ==========
async def setup(bot):
    await bot.add_cog(GameCog(bot))
    # Store awaiting choices
    bot.awaiting_choice = {}
    logger.info("✅ Game cog loaded")

File: game.js
// ONE PIECE - Browser Game Engine
// Cinematic episode-based adventure

// ========== GAME STATE ==========
let gameState = {
    player: {
        name: 'Straw Hat',
        bounty: 0,
        crew: {
            boatBuilder: true,
            ninja: false,
            mapDrawer: false
        },
        rubberPowers: true,
        episode: 1,
        scene: 'start'
    },
    currentEpisode: 1,
    currentScene: null,
    processingChoice: false
};

// ========== EPISODE DATA ==========
const EPISODES = {
    1: {
        title: 'SETTING SAIL',
        location: 'Dawn Island Docks',
        scenes: {
            start: {
                id: 'start',
                narrative: `🌅 **The sun rises over Dawn Island.**

A young pirate wearing a straw hat stands at the dock, the wind catching their coat. Behind them, a sturdy ship bobs gently on the waves.

**🛠️ Boat Builder:** *"Captain! The ship is ready. But... where exactly are we heading?"*

The horizon stretches endlessly. Somewhere out there, the greatest treasure ever known waits.`,
                choices: [
                    { text: '🏴‍☠️ "We hunt the One Piece!"', nextScene: 'fishWarning', statChange: { bounty: 10000 } },
                    { text: '🌊 "To the Grand Line!"', nextScene: 'fishWarning', statChange: { bounty: 5000 } },
                    { text: '🗺️ "First, we need a navigator."', nextScene: 'needNavigator', statChange: {} }
                ]
            },
            fishWarning: {
                id: 'fishWarning',
                narrative: `**🐟 A Fish Soldier emerges from the water!**

*"Turn back, human! The Fish Kingdom claims these waters! The One Piece belongs to our king!"*

**⚔️ COMBAT INITIATED!**

The creature lunges, razor-sharp scales glinting in the sunlight.`,
                choices: [
                    { text: '💪 "Gomu Gomu no Punch!" (Rubber powers)', nextScene: 'victory', statChange: { bounty: 25000 } },
                    { text: '🥷 "Boat Builder, handle this!"', nextScene: 'victory', statChange: { bounty: 15000 } },
                    { text: '🏃 "Retreat to the ship!"', nextScene: 'retreat', statChange: {} }
                ]
            },
            victory: {
                id: 'victory',
                narrative: `**VICTORY!**

Your fist stretches impossibly far, slamming into the Fish Soldier. It spirals into the depths with a terrified shriek.

**⭐ You:** *"That's just a taste. The Fish Kingdom knows we're coming now..."*

The crew cheers. The first battle is won. But the real journey has just begun.`,
                choices: [
                    { text: '➡️ CONTINUE TO EPISODE 2', nextScene: 'episodeEnd', statChange: {} }
                ]
            },
            retreat: {
                id: 'retreat',
                narrative: `**You escape to open sea.**

The Fish Soldier laughs as you sail away.

**🐟 Fish Soldier:** *"Run, little pirate! The Fish King will hear of this!"*

The crew looks worried. The bounty on your head might grow... but the adventure continues.`,
                choices: [
                    { text: '➡️ REGROUP AND CONTINUE', nextScene: 'episodeEnd', statChange: { bounty: -5000 } }
                ]
            },
            needNavigator: {
                id: 'needNavigator',
                narrative: `**🛠️ Boat Builder:** *"Smart thinking, Captain. There's a town on the next island. I hear a legendary map drawer lives there..."*

The sails catch the wind. A new destination appears on the horizon.`,
                choices: [
                    { text: '⛵ "Set course for that town!"', nextScene: 'episodeEnd', statChange: {} }
                ]
            },
            episodeEnd: {
                id: 'episodeEnd',
                narrative: `**📺 END OF EPISODE 1**

*To be continued...*

The crew sails toward the unknown, the Fish Kingdom in pursuit, and somewhere out there... the Map Drawer waits.

**Next Episode:** The Open Sea – New allies, new enemies, and the first clue to the One Piece.`,
                choices: [
                    { text: '🌊 PREPARE FOR EPISODE 2', nextScene: 'nextEpisode', statChange: {} }
                ]
            }
        }
    },
    2: {
        title: 'THE OPEN SEA',
        location: 'Endless Blue',
        scenes: {
            start: {
                id: 'start',
                narrative: `**Days pass on the endless blue.**

The sun beats down. The wind fills your sails. The Fish Kingdom hasn't been seen since the first encounter.

**⭐ You:** *"Land? Anyone?"*

**🛠️ Boat Builder:** *"Nothing yet, Captain. But the sea feels... strange."*

The water churns. Something large swims beneath.`,
                choices: [
                    { text: '🔍 "Everyone on alert!"', nextScene: 'seaKing', statChange: {} },
                    { text: '🎣 "Let\'s fish. I\'m hungry."', nextScene: 'fishCatches', statChange: {} }
                ]
            },
            seaKing: {
                id: 'seaKing',
                narrative: `**🐉 A SEA KING erupts from the water!**

*"ROOOOAR!"* It's massive – bigger than your entire ship!

**🥷 (If recruited):** *"Captain! What do we do?!"*

The creature's eyes lock onto you. This is no ordinary fish.`,
                choices: [
                    { text: '💪 Stretch and punch it!', nextScene: 'seaKingEscape', statChange: { bounty: 50000 } },
                    { text: '🏃 Outrun it!', nextScene: 'seaKingEscape', statChange: {} },
                    { text: '🤝 Try to communicate', nextScene: 'seaKingFriendly', statChange: {} }
                ]
            },
            seaKingEscape: {
                id: 'seaKingEscape',
                narrative: `**After a desperate struggle, you escape!**

The Sea King dives back into the depths, wounded but alive.

**⭐ You:** *"That was too close... We need to get stronger."*

A small island appears on the horizon. Smoke rises from a town.`,
                choices: [
                    { text: '🏝️ Head to the island', nextScene: 'episodeEnd', statChange: {} }
                ]
            },
            seaKingFriendly: {
                id: 'seaKingFriendly',
                narrative: `**🐉 Sea King:** *"...You're not afraid of me?"*

It speaks! This Sea King is intelligent. 

**🐉 Sea King:** *"The Map Drawer you seek... she lives on the hidden island of Cartographia. I will take you there."*`,
                choices: [
                    { text: '🗺️ "Take us there, friend!"', nextScene: 'episodeEnd', statChange: { bounty: 30000 } }
                ]
            },
            fishCatches: {
                id: 'fishCatches',
                narrative: `**⭐ You:** *"Not bad, Boat Builder!"* 

**🐟 Fish Soldier (in the catch):** *"...You caught me. Please don't eat me. I can tell you about the Fish Kingdom..."* 

The Fish Soldier trembles, clearly terrified.`,
                choices: [
                    { text: '🗣️ "Talk. Now."', nextScene: 'fishIntel', statChange: {} },
                    { text: '🍳 "Dinner is dinner."', nextScene: 'episodeEnd', statChange: { bounty: -5000 } }
                ]
            },
            fishIntel: {
                id: 'fishIntel',
                narrative: `**🐟 Fish Soldier:** *"The Fish King wants the One Piece. He has half a map. Your crew has the other half – with the Map Drawer... but you haven't found her yet. Find her before he does!"*

The Fish Soldier dives overboard and disappears.

The crew exchanges worried glances.`,
                choices: [
                    { text: '🏃 "We sail faster!"', nextScene: 'episodeEnd', statChange: {} }
                ]
            },
            episodeEnd: {
                id: 'episodeEnd',
                narrative: `**📺 END OF EPISODE 2**

*To be continued...*

The Map Drawer is out there. The Fish Kingdom is closer than ever.

**Next Episode:** The Map Drawer – A mysterious cartographer joins the crew as the race for the One Piece intensifies.`,
                choices: [
                    { text: '🗺️ CONTINUE TO EPISODE 3 (Coming Soon)', nextScene: 'comingSoon', statChange: {} }
                ]
            },
            comingSoon: {
                id: 'comingSoon',
                narrative: `**🌊 TO BE CONTINUED...**

Episode 3: The Map Drawer is in development!

The crew continues sailing toward their destiny. The Fish Kingdom grows bolder. And somewhere out there... the One Piece waits.

*Thank you for playing!*`,
                choices: [],
                isEnd: true
            }
        }
    }
};

// ========== CANVAS ANIMATION (Cinematic Waves) ==========
const canvas = document.getElementById('gameCanvas');
let ctx = canvas.getContext('2d');

function resizeCanvas() {
    canvas.width = window.innerWidth;
    canvas.height = window.innerHeight;
}
window.addEventListener('resize', resizeCanvas);
resizeCanvas();

// Animated wave effect
let waveOffset = 0;
function drawBackground() {
    if (!ctx) return;
    
    // Gradient sky
    const gradient = ctx.createLinearGradient(0, 0, 0, canvas.height);
    gradient.addColorStop(0, '#0a0a2a');
    gradient.addColorStop(0.5, '#1a3a5a');
    gradient.addColorStop(1, '#0a2a3a');
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 0, canvas.width, canvas.height);
    
    // Draw sun/moon
    ctx.beginPath();
    ctx.arc(canvas.width - 80, 80, 40, 0, Math.PI * 2);
    ctx.fillStyle = '#FFD700';
    ctx.fill();
    ctx.shadowBlur = 20;
    ctx.shadowColor = '#FFD700';
    ctx.fill();
    ctx.shadowBlur = 0;
    
    // Animated waves
    ctx.beginPath();
    for (let x = 0; x <= canvas.width; x += 20) {
        const y = canvas.height - 60 + Math.sin(x * 0.01 + waveOffset) * 15;
        if (x === 0) ctx.moveTo(x, y);
        else ctx.lineTo(x, y);
    }
    ctx.lineTo(canvas.width, canvas.height);
    ctx.lineTo(0, canvas.height);
    ctx.closePath();
    ctx.fillStyle = 'rgba(0, 100, 150, 0.5)';
    ctx.fill();
    
    waveOffset += 0.02;
    requestAnimationFrame(drawBackground);
}

drawBackground();

// ========== UI RENDERING ==========
async function loadScene(episodeNum, sceneId) {
    if (gameState.processingChoice) return;
    
    const episode = EPISODES[episodeNum];
    if (!episode) {
        console.error('Episode not found');
        return;
    }
    
    const scene = episode.scenes[sceneId];
    if (!scene) {
        console.error('Scene not found');
        return;
    }
    
    gameState.currentEpisode = episodeNum;
    gameState.currentScene = scene;
    
    // Hide loading screen, show UI
    document.getElementById('loadingScreen').style.opacity = '0';
    setTimeout(() => {
        document.getElementById('loadingScreen').style.display = 'none';
        document.getElementById('uiOverlay').style.display = 'block';
    }, 1000);
    
    // Update UI
    document.getElementById('episodeTitle').innerHTML = `EPISODE ${episodeNum}: ${episode.title}`;
    document.getElementById('sceneLocation').innerHTML = `📍 ${episode.location}`;
    document.getElementById('narrativeText').innerHTML = scene.narrative;
    document.getElementById('bountyStatus').innerHTML = `${gameState.player.bounty.toLocaleString()} Berries`;
    document.getElementById('episodeStatus').innerHTML = `Ep ${episodeNum}`;
    
    // Count crew
    let crewCount = 1; // Star
    if (gameState.player.crew.boatBuilder) crewCount++;
    if (gameState.player.crew.ninja) crewCount++;
    if (gameState.player.crew.mapDrawer) crewCount++;
    document.getElementById('crewStatus').innerHTML = `Crew: ${crewCount}`;
    
    // Render choices
    const choicesContainer = document.getElementById('choicesContainer');
    choicesContainer.innerHTML = '';
    
    if (scene.choices && scene.choices.length > 0) {
        scene.choices.forEach((choice, index) => {
            const btn = document.createElement('button');
            btn.className = 'choice-btn';
            btn.textContent = choice.text;
            btn.onclick = () => makeChoice(choice);
            choicesContainer.appendChild(btn);
        });
    } else {
        const btn = document.createElement('button');
        btn.className = 'choice-btn';
        btn.textContent = '🏴‍☠️ THE ADVENTURE CONTINUES...';
        btn.onclick = () => {
            if (typeof window.DiscordSDK !== 'undefined') {
                // Notify Discord that episode ended
                console.log('Episode complete - ready for next');
            }
        };
        choicesContainer.appendChild(btn);
    }
}

async function makeChoice(choice) {
    if (gameState.processingChoice) return;
    gameState.processingChoice = true;
    
    // Apply stat changes
    if (choice.statChange) {
        if (choice.statChange.bounty) {
            gameState.player.bounty += choice.statChange.bounty;
        }
    }
    
    // Play sound effect (if any)
    // (Add sound later)
    
    // Determine next scene
    let nextSceneId = choice.nextScene;
    
    // Special handling for next episode
    if (nextSceneId === 'nextEpisode') {
        const nextEp = gameState.currentEpisode + 1;
        if (EPISODES[nextEp]) {
            gameState.player.episode = nextEp;
            await loadScene(nextEp, 'start');
            gameState.processingChoice = false;
            return;
        }
    }
    
    // Load next scene in same episode
    const episode = EPISODES[gameState.currentEpisode];
    const nextScene = episode.scenes[nextSceneId];
    
    if (nextScene) {
        gameState.currentScene = nextScene;
        document.getElementById('narrativeText').innerHTML = nextScene.narrative;
        document.getElementById('bountyStatus').innerHTML = `${gameState.player.bounty.toLocaleString()} Berries`;
        
        // Update choices
        const choicesContainer = document.getElementById('choicesContainer');
        choicesContainer.innerHTML = '';
        
        if (nextScene.choices && nextScene.choices.length > 0) {
            nextScene.choices.forEach((nextChoice) => {
                const btn = document.createElement('button');
                btn.className = 'choice-btn';
                btn.textContent = nextChoice.text;
                btn.onclick = () => makeChoice(nextChoice);
                choicesContainer.appendChild(btn);
            });
        } else {
            const btn = document.createElement('button');
            btn.className = 'choice-btn';
            btn.textContent = '🏴‍☠️ TO BE CONTINUED...';
            choicesContainer.appendChild(btn);
        }
    }
    
    gameState.processingChoice = false;
}

// ========== DISCORD ACTIVITY INTEGRATION ==========
async function initDiscord() {
    // Check if running in Discord
    const isInDiscord = window.location.hostname.includes('discord.com') ||
                        window.location.hostname.includes('discordapp.com') ||
                        window.location.hostname.includes('discordsays.com');
    
    if (isInDiscord && typeof DiscordSDK !== 'undefined') {
        try {
            const discordSdk = new DiscordSDK('YOUR_CLIENT_ID');
            await discordSdk.ready();
            console.log('🎮 Running inside Discord Activity');
            
            // Get user info
            const { code } = await discordSdk.commands.authorize({
                client_id: 'YOUR_CLIENT_ID',
                scope: ['identify']
            });
            
            // Set player name from Discord
            const user = await discordSdk.commands.getUser();
            if (user && user.username) {
                gameState.player.name = user.username;
                document.getElementById('episodeTitle').innerHTML = `⭐ ${user.username}'S JOURNEY`;
            }
        } catch (e) {
            console.log('Discord SDK error:', e);
        }
    }
}

// ========== START GAME ==========
window.addEventListener('DOMContentLoaded', () => {
    setTimeout(() => {
        loadScene(1, 'start');
    }, 2000);
    
    initDiscord();
});